File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    removeDuplicates(nums) {
        let L = 0 ; 
        let R = 0 ; 

        while(R < nums.length){
            if(nums[R]!==nums[L]){
                L++ ; 
                nums[L] = nums[R]
            }R++
        }return L +1
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        

        for(int num: nums){
            seen.add(num);
        }

        int longest = 0; 
        
        for(int num: nums){
            if (!seen.contains(num - 1)) { // only start from the beginning of a sequence
            int curr = num;
            int count = 1;

            while (seen.contains(curr + 1)) {
                curr++;
                count++;
            }

            longest = Math.max(longest, count);
        }
        }

        return longest;
    }
}
